tests/prefsApiTests.js

```diff
@@ -0,0 +1,217 @@
+    // test the /save_prefs API
+    // 1.1 Preparation: go through the login process to generate a login token
+    response = await fluid.tests.utils.sendRequest(serverUrl, "/sso/google", {
+        "headers": {
+            "referer": fluid.tests.utils.mockReferer
+        }
+    });
+    response = await fluid.tests.utils.sendRequest(serverUrl, "/sso/google/login/callback?code=" + fluid.tests.utils.mockAuthCode + "&state=" + fluid.tests.utils.authPayload.state);
+
+    // 1.2 Retrieve the generated login token
+    response = await dbOps.getLoginToken(1, "https://external.site.com");
+
+    const loginToken = response.login_token;
+    const prefsToSave = {
+        textSize: 2
+    };
+
+    // 1.3 Use the login token to test /save_prefs API
+    response = await fluid.tests.utils.sendRequest(serverUrl, "/save_prefs", {
+        headers: {
+            "Authorization": "Bearer " + loginToken
+        }
+    }, "post", prefsToSave);
+    fluid.tests.utils.testResponse(response, 200, "Saved successfully.", "/save_prefs (should succeed)");
+    jqUnit.assertDeepEq("The new preferences object is saved in the database", prefsToSave, await dbOps.getPreferences(loginToken));
+
+    // 2. Failed case: loginToken is not provided
+    response = await fluid.tests.utils.sendRequest(serverUrl, "/save_prefs", null, "post", prefsToSave);
+    fluid.tests.utils.testResponse(response, 403, {"isError": true, "message": "Please login first."}, "/save_prefs");
+
+    // 3. Failed case: wrong loginToken is not provided
+    response = await fluid.tests.utils.sendRequest(serverUrl, "/save_prefs", {
+        headers: {
+            "Authorization": "Bearer wrong"
+        }
+    }, "post", prefsToSave);
+    fluid.tests.utils.testResponse(response, 403, {"isError": true, "message": "Invalid login token. Please login."}, "/save_prefs");
+
+    // 4. Failed case: when the size of the preferences object exceeds the size limit
+    const largeObject = fluid.tests.prefsApi.createLargeObject(2);
+    response = await fluid.tests.utils.sendRequest(serverUrl, "/save_prefs", {
+        headers: {
+            "Authorization": "Bearer " + loginToken
+        }
+    }, "post", largeObject);
+    fluid.tests.utils.testResponse(response, 403, {"isError": true, "message": "The size of preferences exceeds the limit."}, "/save_prefs");
+
+    if (!skipDocker) {
+        // Stop the docker container for the database
+        response = await fluid.personalData.dockerStopDatabase(config.db.dbContainerName);
+        jqUnit.assertTrue("The database docker container has been stopped", response.dbStopped);
+    }
+
+    // Stop the server
+    await server.stopServer(serverInstance.server);
+    serverStatus = await fluid.personalData.getServerStatus(config.server.port);
+    jqUnit.assertFalse("The server has been stopped", serverStatus);
+});
+
+// create a large object: https://gist.github.com/artifi/2047874
```

Comment:
I think this strategy would be more direct: https://stackoverflow.com/a/34811384 e.g.
{
     key: "#".repeat(10240);
}